ttnn/cpp/ttnn/operations/ccl/reduce_scatter/device/host/reduce_scatter_full_worker_grid.cpp

```diff
@@ -406,19 +408,73 @@ static void add_worker_config_to_edm_builders(
                     // Since we are in worker signal EDM termination mode, we don't need to set the actual number of
                     // messages the EDM must forward as it will receive its finish signal from the worker instead
                     1,
-                    receiver_worker_coords,
-                    expected_message_size_bytes);
+                    receiver_worker_coords);
             edm_interface_addresses.worker_receiver_edm_semaphore_addresses.insert(
                 {global_worker_idx, receiver_channel_buffer_info.eth_semaphore_l1_address});
             edm_interface_addresses.worker_receiver_edm_buffer_addresses.insert(
                 {global_worker_idx, receiver_channel_buffer_info.eth_buffer_l1_address});
+            receiver_edm_builder.set_max_message_size_bytes(receiver_channel_buffer_info.channel, clockwise_edm_builders.at(0).get_eth_buffer_size_bytes());
         }
     }
 }
 
-static std::tuple<KernelHandle, KernelHandle> build_reduce_scatter_worker(
+static std::tuple<KernelHandle, KernelHandle, KernelHandle> build_reduce_scatter_worker_ct(
```

Comment:
Why is this change here? Is this the kernel creation optimization but this branch is based off that?

Reply:
Yes. This pr is based on the rt/ct arg optimization pr which has not been merged yet